### ISSUE TITLE
useCollection nested field issue for where clause

### ISSUE THREAD

**Issue by mjbates7** [nandorojo/swr-firestore#53]
Hello,

I have the following code to perform a lookup of all places based on an organisation id that gets passed in but would like to use fields of a nested object (rather than a top level field):
```
interface Place {
  name:string;
  description:string;
  organisation: {
    id: string;
   name: string;
  }
}

export default function Organisation({ props }) {
  const { data, error } = useCollection<Place>(`places`, { 
    where: ['organisation.id', '==', orgId]
  });
```
This code does *run*, however it throws a warning in my IDE: 

`Type '"organisation.id"' is not assignable to type '"name" | "description" | "organisation" | WhereItem<Place, "name" | "description" | "organisation"> | FieldPath'.`

Can you update the `WhereType` (or what is needed) to allow it to use nested properties?

Regards,

Martin

**Comment by nandorojo** [comment]
Interesting. Maybe I could use TypeScript 4.1's template literals for nested fields.

**Comment by mjbates7** [comment]
Further to this, if you use a where clause where you're specifying a nested property, then it makes the type that gets returned not map correctly to the type you've passed in:

![image](https://user-images.githubusercontent.com/4522738/97720114-c91b0900-1abf-11eb-93ee-422bd63dccc6.png)

Any ideas on getting round this?

**Comment by nandorojo** [comment]
Fixed in `0.16.0`
